ffi/src/engine_funcs.rs

```diff
@@ -122,6 +122,7 @@ fn read_parquet_file_impl(
         last_modified: file.last_modified,
         size: file.size,
     };
+    // TODO: Plumb the predicate through the FFI?
```

Comment:
I think 'yes' right? Like the engine can pass a predicate and we can visit it and use it to do skipping etc.

Reply:
Seems complicated enough to be worth a separate PR? The only caller I know of for this method is our toy examples/read-table client, and it doesn't support predicates at all -- the scan is also created without any data skipping predicates. We would need to add some kind of command-line arg parsing support for basic predicates, and then plumb that through to both scan creation and data file reads?

Reply:
Ohh yeah, sorry, I didn't mean in this pr, just that the todo maybe doesn't need a question mark. :)